From 48f374c83d16015d33f208994bec1d09248a5814 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:40:43 -0400
Subject: [alsa] usb-audio: fix BOSS ME-25 MIDI regression

Message-id: <140972823573-415-patchset-jkysela@redhat.com>
Patchwork-id: 91111
O-Subject: [RHEL 7.1 PATCH 414/442] ALSA: usb-audio: fix BOSS ME-25 MIDI regression
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The BOSS ME-25 turns out not to have any useful descriptors in its MIDI
interface, so its needs a quirk entry after all.

Reported-and-tested-by: Kees van Veen <kees.vanveen@gmail.com>
Fixes: 8e5ced83dd1c ("ALSA: usb-audio: remove superfluous Roland quirks")
Cc: <stable@vger.kernel.org>
Signed-off-by: Clemens Ladisch <clemens@ladisch.de>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 53da5ebfef66ea6e478ad9c6add3781472b79475)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/usb/quirks-table.h b/sound/usb/quirks-table.h
index f652b10..223c47b 100644
--- a/sound/usb/quirks-table.h
+++ b/sound/usb/quirks-table.h
@@ -1581,6 +1581,35 @@ YAMAHA_DEVICE(0x7010, "UB99"),
  }
 },
 {
+ /* BOSS ME-25 */
+ USB_DEVICE(0x0582, 0x0113),
+ .driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
+  .ifnum = QUIRK_ANY_INTERFACE,
+  .type = QUIRK_COMPOSITE,
+  .data = (const struct snd_usb_audio_quirk[]) {
+   {
+    .ifnum = 0,
+    .type = QUIRK_AUDIO_STANDARD_INTERFACE
+   },
+   {
+    .ifnum = 1,
+    .type = QUIRK_AUDIO_STANDARD_INTERFACE
+   },
+   {
+    .ifnum = 2,
+    .type = QUIRK_MIDI_FIXED_ENDPOINT,
+    .data = & (const struct snd_usb_midi_endpoint_info) {
+     .out_cables = 0x0001,
+     .in_cables  = 0x0001
+    }
+   },
+   {
+    .ifnum = -1
+   }
+  }
+ }
+},
+{
  /* only 44.1 kHz works at the moment */
  USB_DEVICE(0x0582, 0x0120),
  .driver_info = (unsigned long) & (const struct snd_usb_audio_quirk) {
-- 
1.7.1